src/crypto/equihash.cpp

```diff
@@ -203,6 +205,8 @@ std::set<std::vector<eh_index>> Equihash<N,K>::BasicSolve(const eh_HashState& ba
     X.reserve(init_size);
     for (eh_index i = 0; i < init_size; i++) {
         X.emplace_back(N, base_state, i);
+        // Slow down checking to prevent segfaults (??)
+        if (i % 10000 == 0 && cancelled()) throw solver_cancelled;
```

Comment:
@daira legitimately yes >_< Without this (in the same place in the optimised solver) the miner would consistently segfault after about five seconds of generation. Doesn't happen at any other checkpoint. The comment was so I didn't forget to investigate.